charts/core/src/Echart/useEchart.ts

```diff
@@ -0,0 +1,389 @@
+import { getDefaultChartOptions } from '../Chart/config';
+import * as updateUtils from '../Chart/hooks/updateUtils';
+
+import {
+  EChartHookProps,
+  EChartsInstance,
+  ZoomSelectionEvent,
+} from './Echart.types';
+
+// Singleton promise for initialization to prevent duplication
+let initPromise: Promise<void> | null = null;
+let echartsCore: any;
+let echartsCharts: any;
+let echartsRenders: any;
+let echartsComponents: any;
+
+async function initializeEcharts() {
+  // If already initialized, return immediately
+  if (echartsCore) {
+    return;
+  }
+
+  // If initialization is in progress, wait for it
+  if (initPromise) {
+    return initPromise;
+  }
+
+  // Start initialization and store the promise
+  initPromise = (async () => {
+    try {
+      const [
+        echartsCoreResolved,
+        echartsChartsResolved,
+        echartsRendersResolved,
+        echartsComponentsResolved,
+      ] = await Promise.all([
+        import('echarts/core'),
+        import('echarts/charts'),
+        import('echarts/renderers'),
+        import('echarts/components'),
+      ]);
+
+      echartsCore = echartsCoreResolved;
+      echartsCharts = echartsChartsResolved;
+      echartsRenders = echartsRendersResolved;
+      echartsComponents = echartsComponentsResolved;
+
+      echartsCore.use([
+        echartsCharts.LineChart,
+        echartsRenders.CanvasRenderer,
+        echartsComponents.TitleComponent,
+        echartsComponents.TooltipComponent,
+        echartsComponents.GridComponent,
+        echartsComponents.LegendComponent,
+        echartsComponents.ToolboxComponent,
+        echartsComponents.DataZoomComponent,
+        echartsComponents.DataZoomInsideComponent,
+      ]);
+    } catch (error) {
+      // Ensure we clear the promise if initialization fails
+      initPromise = null;
+      throw error;
+    }
+  })();
+
+  return initPromise;
+}
+
+/**
+ * Wrapper around the ECharts library. Instantiates an ECharts instance.
+ * Provides helper methods to hide ECharts specific logic and give a cleaner API
+ * for interacting with a chart.
+ */
+export function useEchart({
+  container,
+  initialOptions,
+  theme,
+}: EChartHookProps): EChartsInstance {
+  const [echartsInstance, setEchartsInstance] = useState<any>(null); // has to be any since no types exist until import
+  const [error, setError] = useState<EChartsInstance['error']>(null);
+  const [ready, setReady] = useState<EChartsInstance['ready']>(false);
+  const [options, setOptions] = useState<EChartsInstance['options']>(
+    initialOptions || {},
+  );
+
+  // Keep track of active handlers
+  const activeHandlers = useRef(new Map());
+
+  const withInstanceCheck = <T extends (...args: Array<any>) => void>(
+    fn: T,
+  ) => {
+    return (...args: Parameters<T>) => {
+      if (!echartsInstance) {
+        console.error('Echart instance not initialized');
+        return;
+      }
+      fn(...args);
+    };
+  };
```

Comment:
I went back and forth on whether this should actually wait for an instance. In theory, we could store a promise in a ref, and make all of the methods asynchronous. I ultimately decided not to and instead essentially do nothing if the instance doesn't exist yet. But would be curious to here other thoughts.

Reply:
I'm a little confused why this is necessary. Is it just for the error logging? 
Unless I'm missing something, the optional chain check (e.g. L114) should catch if the `echartsInstance` singleton is undefined/null

Reply:
If this were to wait for the instance, that would mean all the instance methods (like `addSeries`) would all be async?

Reply:
> Edit: I guess it's useful to fully abort if the instance isn't defined
That's correct. This is used as a higher order function to wrap any function that relies on the existence of an instance. This is an extraction of duplicate logic that was being used inline in each of those methods, attempting to handle the case where it doesn't exist yet gracefully. 

> If this were to wait for the instance, that would mean all the instance methods (like addSeries) would all be async?
That's correct. It would basically wrap the given function in a promise, converting it to an async function that's capable of being awaited, so instead of just (almost) silently ignoring this case, the execution wouldn't occur until the instance did exist.

Reply:
Fwiw, I don't think I like this. I think the interface of just exposing some way to determine readiness is fine.